### PR TITLE
Zip for state validation on patient profile

### PR DESCRIPTION
**Pull request by mehansen**
# FRONTEND PULL REQUEST

## Related Issue

- issue #3202 

## Changes Proposed

- Validate ZIP code matches state in Person form on save
- added test case to AddPatient test

## Additional Information

- tried to add ZIP validation to the yup schema but ran into issues with how often we validate against it
- followup issue #3595 to address including this validation inline with the form field, to be more like other form errors
- only for US

## Testing

Navigate to add patient, edit patient page, or patient self-registration. If zip is invalid for state and all other fields are valid, invalid zip error toast will appear when user saves. If there are other form errors, the zip toast will not appear. If all fields are valid, address validation popup will appear as normal.

## Screenshots / Demos

Reviewed by design team in 3/29 design sync meeting

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes have been approved by content team

### Support
- [ ] Any changes that might generate new support requests have been flagged to the support team
- [ ] [n/a] Any changes to support infrastructure have been demo'd to support team

### Security
- [ ] [n/a] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] [n/a] Any dependencies introduced have been vetted and discussed

### CODE REVIEW

File: frontend/src/lang/es.ts [hunk 1]
```diff
@@ -195,6 +195,7 @@ export const es: LanguageConfig = {
           street: "Falta el nombre de la calle",
           streetTwo: "Calle 2 tiene un formato incorrecto",
           zipCode: "Falta el código postal o tiene un formato incorrecto",
+          zipForState: "Código postal no válido para este estado",
```
Comment (mehansen) [inline frontend/src/lang/es.ts:198]: @Rebecca-LM can you please review this translation?

Comment (Rebecca-LM) [reply frontend/src/lang/es.ts:198]: @mehansen I'd change to: "El código postal no es válido para este estado"

File: frontend/src/app/patients/AddPatient.test.tsx [hunk 2]
```diff
@@ -352,7 +360,59 @@ describe("AddPatient", () => {
...
+          screen.queryAllByText("Save Changes", {
+            exact: false,
+          })[0]
+        );
+        expect(
+          await screen.findByText("Invalid ZIP code for this state", {
+            exact: false,
+          })
+        ).toBeInTheDocument();
+      });
```
Comment (nathancrtr) [inline frontend/src/app/patients/AddPatient.test.tsx:415]: Great test ✌️

**Review by nathancrtr — DISMISSED** [review]
LGTM. Definitely don't blame you for punting on integrating this with `yup`, it's a pain.

**Review by nickclyde — DISMISSED** [review]
Looks great! Awesome work Merethe 😄

**Review by nickclyde — APPROVED** [review]

**Review by nathancrtr — APPROVED** [review]
Re-approving after review was dismissed.

**Pull request merged**